sycl/include/sycl/ext/oneapi/matrix/matrix-jit-use.hpp

```diff
@@ -0,0 +1,517 @@
+//==------------------ matrix-jit-use.hpp - SYCL matrix ----------------*- C++
+//-*---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===--------------------------------------------------------------------=== //
+
+#pragma once
+
+#include <CL/__spirv/spirv_ops.hpp>
+#include <CL/sycl/detail/defines_elementary.hpp>
+#include <CL/sycl/feature_test.hpp>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace experimental::matrix {
+
+// packed_a and packed_b will be replaced by packed once the use implementation
+// is stable.
+enum class matrix_layout { row_major, col_major, packed_a, packed_b };
+
+template <matrix_layout Layout> struct spv_matrix_layout_traits {
+  static constexpr __spv::MatrixLayout value = __spv::MatrixLayout::RowMajor;
+};
+
+#define SPV_MATRIX_LAYOUT_TRAITS(LAYOUT, SPV_LAYOUT)                           \
+  template <> struct spv_matrix_layout_traits<LAYOUT> {                        \
+    static constexpr __spv::MatrixLayout value = SPV_LAYOUT;                   \
+  };
+
+SPV_MATRIX_LAYOUT_TRAITS(matrix_layout::row_major,
+                         __spv::MatrixLayout::RowMajor)
+SPV_MATRIX_LAYOUT_TRAITS(matrix_layout::col_major,
+                         __spv::MatrixLayout::ColumnMajor)
+SPV_MATRIX_LAYOUT_TRAITS(matrix_layout::packed_a, __spv::MatrixLayout::PackedA)
+SPV_MATRIX_LAYOUT_TRAITS(matrix_layout::packed_b, __spv::MatrixLayout::PackedB)
+
+// unnecessary was introduced for backward compatibility.
+// Once the use implementation is stable, "unnecessary" value will be omitted
+enum class matrix_use { matrix_a, matrix_b, accumulator, unnecessary };
```

Comment:
in matrix-tensorcores.hpp we define "a"/"b" rather than "matrix_a"/"matrix_b":
```
enum class matrix_use { a, b, accumulator };
```
We should align on a single naming for matrix_use, since it is a user facing class that the user will have to provide to the joint_matrix template constructor.